src/zmq/zmqpublishnotifier.cpp

```diff
@@ -230,6 +232,13 @@ bool CZMQPublishHashInstantSendDoubleSpendNotifier::NotifyInstantSendDoubleSpend
         && SendMessage(MSG_HASHISCON, dataPreviousHash, 32);
 }
 
+bool CZMQPublishHashRecoveredSigNotifier::NotifyRecoveredSig(const llmq::CRecoveredSig &sig)
+{
+    LogPrint(BCLog::ZMQ, "zmq: Publish hashrecoveredsig %s\n", sig.msgHash.ToString());
+    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
+    ss << sig.msgHash;
```

Comment:
Shouldn't this be more like
```suggestion
    uint256 hash = sig.GetHash();
    LogPrint(BCLog::ZMQ, "zmq: Publish hashrecoveredsig %s\n", hash.ToString());
    CDataStream ss(SER_NETWORK, PROTOCOL_VERSION);
    ss << hash;
```
?

Reply:
Hmm thats how i implemented it initially but then i realized  `CZMQPublishHashChainLockNotifier::NotifyChainLock` doesn't send the hash of the chainlock itself but the hash of the locked block and `CZMQPublishHashTransactionLockNotifier::NotifyTransactionLock` doesn't send the hash of the islock but the hash of the locked tx. So i just aligned the behaviour of this one with those two. And it feels like it makes more sense because you should know `msgHash` it in advance and you could use it as some "message x was singed" trigger or so... only issue i see here is that the signature of `msgHash` also depends on `id` so its probably not 100% guaranteed to be unique this way... Thoughts?

Reply:
I see. Makes sense especially now that I realized that `quorum getrecsig` rpc actually requires `msgHash` and not sig hash :) Still feels like we are kind of mis-using `hash` prefix here (and in other similar cases you mentioned which was why I proposed https://github.com/dashpay/dash/issues/3713#issuecomment-694757577) but that's probably fine for now and should be fixed together with other "broken" messages in a different PR.